sunpy/physics/differential_rotation.py

```diff
@@ -84,138 +89,315 @@ def diff_rot(duration: u.s, latitude: u.deg, rot_type='howard', frame_time='side
 
     A, B, C = rot_params[rot_type]
 
+    # This calculation of the rotation assumes a sidereal frame time.
     rotation = (A + B * sin2l + C * sin4l) * duration
 
+    # Applying this correction assumes that the observer is on the Earth,
+    # and that the Earth is at the same distance from the Sun at all times
+    # during the year.
     if frame_time == 'synodic':
         rotation -= 0.9856 * u.deg / u.day * duration
 
     return Longitude(rotation.to(u.deg))
 
 
-def solar_rotate_coordinate(coordinate,
-                            new_observer_time,
-                            new_observer_location="earth",
-                            **diff_rot_kwargs):
+def _get_new_observer(initial_obstime, observer, time):
+    """
+    Helper function that interprets the possible ways of specifying the
+    input to the solar coordinate rotation function.
+
+    If the "observer" argument is not None, it is used to specify the location
+    of the new observer in space and time.
+
+    If the "time" argument is not None, it is used to calculate the duration
+    over which to the amount of solar rotation is calculated. Note that using
+    the "time" keyword assumes that the new observer is on the Earth. This may
+    be a reasonable assumption depending on the application.
+
+    Either the "observer" or "time" argument must be specified, but both
+    cannot be specified at the same time and both cannot be None.
+
+    Parameters
+    ----------
+    initial_obstime : `~astropy.time.Time`
+        The initial time before solar rotation has been applied.
+
+    observer : `~astropy.coordinates.BaseCoordinateFrame`, `~astropy.coordinates.SkyCoord`, None
+        The location of the new observer in space and time (the observer must have an
+        interpretable obstime property).
+
+    time : `~astropy.time.Time`, `~astropy.time.TimeDelta`, `~astropy.units.Quantity`, None
+        Used to define the duration over which the amount of solar rotation is
+        calculated.  If 'time' is an `~astropy.time.Time` then the time interval is
+        "time - initial_obstime"; if 'time' is `~astropy.time.TimeDelta` or
+        `~astropy.units.Quantity` then the calculation is "initial_obstime + time".
+
+    Returns
+    -------
+    new_observer : `~astropy.coordinates.SkyCoord`
+        The position of the observer in space and time after solar rotation has been
+        applied.
+    """
+    # Check the input and create the new observer
+    if (observer is not None) and (time is not None):
+        raise ValueError("Either the 'observer' or the 'time' keyword must be specified, but not both simultaneously.")
+    elif observer is not None:
+        # Check that the new_observer is specified correctly.
+        if not (isinstance(observer, (BaseCoordinateFrame, SkyCoord))):
```

Comment:
The output type of this function is dependant on the input type. If it is called with a SkyCoord it will output one and if called with a frame it will output a frame. Preferably this function should be changed to always output one or the other (I think frame is the output of `get_body`? if so frame would probably be easier?) or at least the docstring should be updated.

Reply:
I took the easy path and fixed the docstring.